packages/nimble/R/cppDefs_cppProject.R

```diff
@@ -228,8 +223,27 @@ cppProjectClass <- setRefClass('cppProjectClass',
                                        if(status != 0) 
                                            stop(structure(simpleError("Failed to create the shared library"),
                                                           class = c("SHLIBCreationError", "ShellError", "simpleError", "error", "condition")))
-                                       nimbleUserNamespace$sessionSpecificDll <- dyn.load(ssDllName, local = TRUE)
-                                   },                                  
+                                       return(dyn.load(ssDllName, local = TRUE))
+                                   },
+                                   compileDynamicRegistrations = function(showCompilerOutput = nimbleOptions('showCompilerOutput')) {
+                                       timeStamp <- format(Sys.time(), "%m_%d_%H_%M_%S")
+                                       dllName <- paste0("dynamicRegistrations_", timeStamp)
+                                       cppName <- paste0(dllName, ".cpp")
+                                       writeDynamicRegistrationsDotCpp(dynamicRegistrationsCppName, dynamicRegistrationsDllName)
+                                       nimbleUserNamespace$sessionSpecificDll <- compileStaticCode(dllName, cppName)
+                                   },
+                                   compileTensorflowWrapper = function(showCompilerOutput = nimbleOptions('showCompilerOutput')) {
+                                       # Load prerequisite DLLs.
+                                       library(tensorflow)
+                                       tfPath <- tf$`__path__`  # This line triggers loading of _pywrap_tensorflow_internal.so.
+                                       nimbleUserNamespace$tensorflowDll <- file.path(tfPath, 'python', '_pywrap_tensorflow_internal.so')
```

Comment:
Usually when we name a variable as a 'XYZDll', we mean it is an object of class `DLLinfo`, typically returned by `dyn.load`.  E.g. that's what happens returning from (your refactor of) `compileStaticCode`.  One main purpose in hanging on to that object is to use it in `.Call( <DLLinfo object>$<registered name>, ...`). 

I don't know what mechanism `tf$...` uses to load its DLL and I guess we don't need to worry about using `.Call` with it directly since we'll go through its R API.

So maybe the fix is simply to rename `tensorFlowDll` to `tensorFlowDllFile`?  

Do we definitely need our hands on it later or is the purpose limited to error-trapping on whether the file exists?  The only reason to put it in `nimbleUserNamespace` is if we need it in other places.